cloudcix_primitives/cloudinit_lxd.py

```diff
@@ -0,0 +1,225 @@
+    """ Retrieve cloud-init configuration from an LXD container.
+    
+    :param endpoint_url: The endpoint URL for the LXD Host.
+    :param project: The LXD project name.
+    :param container_name: The name of the LXD container.
+    :param config_type: Type of cloud-init config ('user-data', 'vendor-data', or 'network-config').
+    :param verify_lxd_certs: Boolean to verify LXD certs.
+    :return: A tuple with a boolean flag indicating success or failure, a dictionary containing the configuration, and a message.
+    """
+    # Validate config_type
+    valid_types = ('user-data', 'vendor-data', 'network-config')
+    if config_type not in valid_types:
+        return False, {}, f"Invalid config_type: {config_type}. Must be one of: {', '.join(valid_types)}"
+    
+    # Define message
+    config_key = f"cloud-init.{config_type}"
+    messages = {
+        1000: f'Successfully retrieved cloud-init {config_type} from container {container_name} on {endpoint_url}',
+        1001: f'No cloud-init {config_type} configuration found for container {container_name} on {endpoint_url}',
+        3021: f'Failed to connect to {endpoint_url} for containers.get payload',
+        3022: f'Failed to run containers.get payload on {endpoint_url}. Payload exited with status ',
+    }
+
+    def run_host(endpoint_url, prefix, successful_payloads):
+        rcc = LXDCommsWrapper(comms_lxd, endpoint_url, verify_lxd_certs, project)
+        fmt = HostErrorFormatter(
+            endpoint_url,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads,
+        )
+        
+        result = {}
+        
+        # Get the instance
+        ret = rcc.run(cli='containers.get', name=container_name)
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+1}: {messages[prefix+1]}"), fmt.successful_payloads, result
+        if ret["payload_code"] != API_SUCCESS:
+            return False, fmt.payload_error(ret, f"{prefix+2}: {messages[prefix+2]}"), fmt.successful_payloads, result
+
+        instance = ret['payload_message']
+        fmt.add_successful('containers.get', ret)
+        
+        # Get the cloud-init configuration
+        cloud_init_config = instance.config.get(config_key, '')
+        
+        if cloud_init_config:
+            result[config_key] = cloud_init_config
+            fmt.add_successful('cloud_init.read', {config_key: 'found'})
+            message = f'1000: {messages[1000]}'
+        else:
+            result[config_key] = None
+            fmt.add_successful('cloud_init.read', {config_key: 'not_found'})
+            message = f'1001: {messages[1001]}'
+            
+        return True, message, fmt.successful_payloads, result
+
+    status, msg, successful_payloads, result = run_host(endpoint_url, 3020, {})
+    
+    if status is False:
+        return False, {}, msg
+
+    return True, result, msg
+
+def scrub(
```

Comment:
I do not think scrub of the user-data cloudinit file is required as I cannot think of a scenario where it will be required